src/engine/PhotoStudio.cs

```diff
@@ -39,10 +39,16 @@ public partial class PhotoStudio : SubViewport
     private readonly Dictionary<ISimulationPhotographable.SimulationType, IWorldSimulation> worldSimulations = new();
     private readonly Dictionary<IWorldSimulation, Node3D> simulationWorldRoots = new();
 
-    private readonly Queue<ImageTask> tasks = new();
+    private readonly PriorityQueue<ImageTask, int> tasks = new();
```

Comment:
Using `PriorityQueue<ImageTask, int>` with `taskPriorityOffset` looks unfancy to me. I think it's not a good idea to mix an offset priority with an index number of a task. I think the code would be better if you used `pair<int, int>` or a small structure with a Compare function. The first value should be the priority; the second value should be the index number.

Reply:
I actually agree with this proposal. This is a cleaner approach, especially considering that without it, tasks of different priorities might have an incorrect order (in some edge cases).
According to the [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.tuple-2.system-icomparable-compareto?view=net-8.0), (int, int) tuples seem to be compared in the such way, that the second ints are used for comparison only when the first ones are equal, which is pretty much what we need here.

Reply:
I don't *fully* agree because in a case where a higher priority level is absolutely filled with tasks, it'd be extremely nice to still rarely allow lower priority items to get through (though in this case it would be beneficial to make newer tasks more important so that for example if the player a minute ago looked at some high priority images and no goes elsewhere in the game, then that other place might take multiple minutes to load rather than get to be in front of the queue).

I still agree that this current PR is an improvement and is probably fine for now.